pkg/graph/resolvers/system_intake_status_admin.go

```diff
@@ -0,0 +1,96 @@
+func SystemIntakeStatusAdminGet(intake *models.SystemIntake) (models.SystemIntakeStatusAdmin, error) {
+
+	if intake.State == models.SystemIntakeStateCLOSED && intake.DecisionState == models.SIDSNoDecision { // If the decision is closed and a decision wasn't issued, show closed
+		return models.SISAClosed, nil
+
+	}
+	var retStatus models.SystemIntakeStatusAdmin
+	var err error
+	switch intake.Step {
+	case models.SystemIntakeStepINITIALFORM:
+		retStatus = calcSystemIntakeInitialFormStatusAdmin(intake.RequestFormState)
+	case models.SystemIntakeStepDRAFTBIZCASE:
+		retStatus = calcSystemIntakeDraftBusinessCaseStatusAdmin(intake.DraftBusinessCaseState)
+	case models.SystemIntakeStepGRTMEETING:
+		retStatus = calcSystemIntakeGRTMeetingStatusAdmin(intake.GRTDate)
+	case models.SystemIntakeStepFINALBIZCASE:
+		retStatus = calcSystemIntakeFinalBusinessCaseStatusAdmin(intake.FinalBusinessCaseState)
+	case models.SystemIntakeStepGRBMEETING:
+		retStatus = calcSystemIntakeGRBMeetingStatusAdmin(intake.GRBDate)
+	case models.SystemIntakeStepDECISION:
+		retStatus, err = calcSystemIntakeDecisionStatusAdmin(intake.DecisionState)
+	default:
+		return retStatus, fmt.Errorf("issue calculating the admin state status, no valid step")
+
+	}
+	return retStatus, err
+}
+
+func calcSystemIntakeInitialFormStatusAdmin(intakeFormState models.SystemIntakeFormState) models.SystemIntakeStatusAdmin {
+	if intakeFormState == models.SIRFSSubmitted {
+		return models.SISAInitialRequestFormSubmitted
+	}
+	return models.SISAInitialRequestFormInProgress
+}
+
+func calcSystemIntakeDraftBusinessCaseStatusAdmin(draftBuisnessCaseState models.SystemIntakeFormState) models.SystemIntakeStatusAdmin {
+	if draftBuisnessCaseState == models.SIRFSSubmitted {
+		return models.SISADraftBusinessCaseComplete
+	}
+	return models.SISADraftBusinessCaseInProgress
+}
+
+func calcSystemIntakeGRTMeetingStatusAdmin(grtDate *time.Time) models.SystemIntakeStatusAdmin {
+
+	if grtDate == nil {
+		return models.SISAGrtMeetingReady
+	}
+
+	if grtDate.After(time.Now()) {
+		return models.SISAGrtMeetingReady
+	}
+	return models.SISAGrtMeetingComplete
+}
+
+func calcSystemIntakeFinalBusinessCaseStatusAdmin(finalBuisnessCaseState models.SystemIntakeFormState) models.SystemIntakeStatusAdmin {
+	if finalBuisnessCaseState == models.SIRFSSubmitted {
+		return models.SISAFinalBusinessCaseComplete
+	}
+	return models.SISAFinalBusinessCaseInProgress
+}
+
+func calcSystemIntakeGRBMeetingStatusAdmin(grbDate *time.Time) models.SystemIntakeStatusAdmin {
+
+	if grbDate == nil {
+		return models.SISAGrbMeetingReady
+	}
+
+	if grbDate.After(time.Now()) {
+		return models.SISAGrbMeetingReady
+	}
+	return models.SISAGrbMeetingComplete
+}
+
+func calcSystemIntakeDecisionStatusAdmin(decisionStatus models.SystemIntakeDecisionState) (models.SystemIntakeStatusAdmin, error) {
```

Comment:
It's weird that only this function throws an error. So basically if the `step` is `models.SystemIntakeStepDECISION`, the state must be one of LCID issued, not approved, or not a governance request?

I know we argued over this before, but I would rather not check for an invalid state and instead make that state an impossibility. Would it not be better to forgo checking the decision step and check for a closed state and return the type of decision before the switch statement?  

```
if (intake.State == models.SystemIntakeStateCLOSED && intake.DecisionState == models.SIDSNoDecision) {
  return models.SISAClosed
} else if (intake.State == models.SystemIntakeStateCLOSED && intake.DecisionState != models.SIDSNoDecision) {
  return calcSystemIntakeDecisionStatusAdmin(intake.DecisionState)
}

// switch for steps
```

This way if a request is re-opened, it just defaults to the previous step the intake was on and prevents the invalid state.

We could also return `models.SISAClosed` from `calcSystemIntakeDecisionStatusAdmin()` and just have: 

```
if (intake.State == models.SystemIntakeStateCLOSED) {
  return calcSystemIntakeDecisionStatusAdmin(intake.DecisionState)
}
// switch for steps
```

As far as I can tell, the only thing having the explicit decision step buys us is the possibility of having `open` and `LCID issued` or some other combination when a request is re-opened. However, I question the utility of that state. It would be changed as soon as the admin moves the request to a new step, which I think they would immediately do since the decision step has no associated actions for the requester.

I also question if we should set the decision state to `NO_DECISION` when a request is re-opened, since I don't think the UI supports "re-close" with the previous decision. If we need the previously issued LCID, that's already stored on the intake table, and no other data is required for the other 3 close options.

Reply:
According to [Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=331-105231&mode=design&t=a1FVWMC5kaIrDcJU-0), it looks like you can close a request that you previously opened with the same decision (it doesn't care that it was re-opened, it just cares that it is currently open).

While I do like the idea of not exposing an error, I don't think you can remove `SystemIntakeStepDECISION` without re-designing the view that is requested by figma. 

I don't think we need to remove the step, but if we did get approval to change the approach, and the representation in figma, it should happen in a separate ticket.  This ticket's scope is just for the admin status.

Reply:
> According to [Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=331-105231&mode=design&t=a1FVWMC5kaIrDcJU-0), it looks like you can close a request that you previously opened with the same decision (it doesn't care that it was re-opened, it just cares that it is currently open).

> While I do like the idea of not exposing an error, I don't think you can remove SystemIntakeStepDECISION without re-designing the view that is requested by figma.
I agree with this, although I think it's worth talking with UX about what the expected result of re-opening in that case should look like.

> I know we argued over this before, but I would rather not check for an invalid state and instead make that state an impossibility. Would it not be better to forgo checking the decision step and check for a closed state and return the type of decision before the switch statement?
I'm not sure if that'd improve the code for `calcSystemIntakeDecisionStatusAdmin`; either that code has explicit checks for every case except where `intake.DecisionState` is `models.SIDSNoDecision`, like it is now (in which case it needs to have the possibility of returning an error), or there's a default case as with the other private functions here (which would return an erroneous result if `calcSystemIntakeDecisionStatusAdmin()` got called with `intake.DecisionState == models.SIDSNoDecision`).

We can't exclude invalid states fully statically (at least not with the current data model). If we did get rid of decision as an explicit step, that would help, but given the Figma that doesn't seem like an option right now. Given those constraints, I think this logic is about as good as we can do for now.

All that said, I think it'd be worth not having `calcSystemIntakeDecisionStatusAdmin()` be the only private function that returns an error. We could either have it default to returning `models.SISANotApproved` (with a comment that the only three valid outcomes of a decision at this point are `LcidIssued`, `NoGovernance`, and `NotApproved`), or we could have all functions explicitly check for possible invalid states and return errors in those cases (i.e. if we somehow have `intake.Step == models.SystemIntakeStepDRAFTBIZCASE`, call `calcSystemIntakeDraftBusinessCaseStatusAdmin()`, but `draftBusinessCaseState == `SIRFSNotStarted`).

Reply:
Update, we did a sync with Zoë, and it makes the most sense to keep `DECISION_AND_NEXT_STEPS` as a step. As further attempt to harden the code against improper state, I introduced a new migration with a check constraint.

 I think that this along with returning an error for an invalid state should help ensure we don't return invalid state.

Reply:
@StevenWadeOddball That works for me. What do you think about the idea of having either `calcSystemIntakeDecisionStatusAdmin()` return `models.SISANotApproved` as default (and thus not returning an error) or having all functions check for invalid states and possibly returning errors? Think it's worth making a change?

Reply:
I'm not quite sure which way to go on returning errors from the status. On the one hand, it's probably good to throw an error because it will be very obvious that we are setting that state incorrectly, but on the other, it feels unnecessary and complicates this code and everything that calls it that now has to check for that returned error.

Reply:
@mynar7 I feel about the same, I think both have merit. I lean towards not returning an error, but it's a very slight preference.

Reply:
Thanks @DylanSpOddball and @mynar7, I agree, I think both have merit. I decided to leave the error handling as is for now.  I do think it would be nice to not need to show an error, but I do like that this explicitly showing this is an error.

 I'm ok if we end up deciding to change it later to fit a different paradigm. I'd especially like if this didn't need to show an error at all, but I think this makes sense considering there is a potential invalid state combination.